fasort.sh

```diff
@@ -2,21 +2,21 @@
 # Author: Noxy (a dhole who likes paws)
 # This script will find and move any FA files in the working directory and move them to artist folders under $SORTDIR.
 
+set -euo pipefail
+
 # Directory where you want files to end up
 SORTDIR=~/Pictures/FA_Sorted
 
 # Find files in pwd that match what I think is a decent regex for FA image files
-for f in `ls | egrep '^\d+\.[0-9a-zA-Z-]+_.+\.\w+$'`
+find . -iregex '.*/[0-9]+\.[a-z0-9]+_.*' -print0 | while read -d $'\0' f
 do
 
     # awk out the artist name
-    ARTIST=$(echo $f | awk -F '.' '{print$2}' | awk -F _ '{print$1}')
+    ARTIST="$(printf '%s' "$(basename "$f")" | awk -F '.' '{print$2}' | awk -F _ '{print$1}')"
 
     # Create the directory for the artist name if it doesn't exist already
-    if [ ! -d $SORTDIR/$ARTIST ]; then
-        mkdir -p $SORTDIR/$ARTIST
-    fi
+    mkdir -p "$SORTDIR/$ARTIST"
```

Comment:
`mkdir -p` is already a no-op when the directory already exists, no need to check it beforehand